### PR TITLE
HW4 done

### PR DESCRIPTION
**Pull request by foria1405**
#### New profile:
* AllTestsSeleniumHW4

Example: 
mvn test -P AllTestsSeleniumHW4 
mvn allure:report 
mvn allure:serve

#### Result
Tests run: 3, Failures: 1, Errors: 0, Skipped: 0

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -30,60 +39,97 @@
...
+            </plugin>
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-surefire-plugin</artifactId>
-                <version>3.0.0-M5</version>
+                <version>${maven-surefire-plugin.version}</version>
                 <configuration>
                     <suiteXmlFiles>
-                        <suiteXmlFile>${testng.suite.file}</suiteXmlFile>
+                        ${maven-surefire-plugin.suiteXmlFile}/AllTestsSeleniumHW4.xml
```
Comment (akulebyakin) [inline pom.xml]: Why `.../AllTestsSeleniumHW4.xml` hardcoded here?

File: pom.xml [hunk 2]
```diff
@@ -92,48 +138,80 @@
...
         </profile>
 
+        <profile>
+            <id>AllTestsSeleniumHW4</id>
+            <properties>
+                <maven-surefire-plugin.suiteXmlFile>
+                    ${maven-surefire-plugin.suiteXmlFile}/AllTestsSeleniumHW4.xml
+                </maven-surefire-plugin.suiteXmlFile>
+            </properties>
+        </profile>
```
Comment (akulebyakin) [inline pom.xml]: Here you're trying to define property `maven-surefire-plugin.suiteXmlFile` using that property inside. It doesn't work. Maven fails with `NullPointerException` on this step.
Before you had 2 properties: `testng.suite.file` and `testng.suite.path` and it worked.

File: pom.xml [hunk 3]
```diff
@@ -30,60 +39,97 @@
...
+                        ${maven-surefire-plugin.suiteXmlFile}/AllTestsSeleniumHW4.xml
                     </suiteXmlFiles>
+                    <argLine>
+                        -javaagent:"${settings.localRepository}/org/aspectj/aspectjweaver/${aspectj.version}/aspectjweaver-${aspectj.version}.jar"
+                    </argLine>
+                    <systemProperties>
+                        <property>
+                            <name>allure.results.directory</name>
+                            <value>${basedir}/target/allure-results</value>
+                        </property>
```
Comment (akulebyakin) [inline pom.xml]: Property `allure.results.directory` defined 3 times in the project:
- pom.xml -> maven-surefire-plugin -> systemProperties
- pom.xml -> reporting -> allure-maven -> configuration
- allure.properties

Please leave only one property definition

File: src/test/resources/allure/allure.properties [hunk 4]
```diff
@@ -0,0 +1 @@
+allure.results.directory=target/allure-results
```
Comment (akulebyakin) [inline src/test/resources/allure/allure.properties]: If you move `allure.properties` from `src/test/resources/allure` directory to `src/test/resources` this configuration will work.

**Review by akulebyakin — CHANGES_REQUESTED** [review]
Please review and resolve comments and fix CheckStyle warnings

File: pom.xml (hunk 1 above)
Comment (foria1405) [reply pom.xml]: Remove hardcoded .../AllTestsSeleniumHW4.xml

File: pom.xml (hunk 2 above)
Comment (foria1405) [reply pom.xml]: Removed maven-surefire-plugin.suiteXmlFile. Returned testng.suite.file and testng.suite.path

File: pom.xml (hunk 3 above)
Comment (foria1405) [reply pom.xml]: Left only one property definition

File: src/test/resources/allure/allure.properties (hunk 4 above)
Comment (foria1405) [reply src/test/resources/allure/allure.properties]: Moved allure.properties from src/test/resources/allure directory to src/test/resources

**Review by akulebyakin — APPROVED** [review]
Approved!

**Pull request merged**